draft-pda-protocol.md

```diff
@@ -573,116 +612,288 @@ PA protocol. The structure of the leader share is similarly protocol specific:
 {{pa-error-common-aborts}}.
 
-[TODO: Since we're running the protocol with multiple cohorts of aggregators, the
-collector needs to decide how to pick which cohort has the "correct" output.
-This might be the cohort with the largest batch of inputs. Figure this out once
-the collection is specified.]
+## Collect {#pa-collect}
+
+~~~~
+Collector     Leader           Helper
+  |  collect 1  |                |
+  +------------->                |
+  |             |  aggregate 1   |
+  |             <---------------->
+  |             |  ...           |
+  |             |  aggregate L   |
+  |             <---------------->
+  |             |  output share  |
+  |             <---------------->
+  <-------------+                |
+  |  ...        |                |
+  |  collect N  |                |
+  +------------->                |
+  |             |  aggregate 1   |
+  |             <---------------->
+  |             |     ...        |
+  |             |  aggregate L   |
+  |             <---------------->
+  |             |  output share  |
+  |             <---------------->
+  <-------------+                |
+  v             v                v
+~~~~
+{: #pa-collect-flow title="Flow of the collect process with N collect requests
+and L aggregate requests per collect request."}
+
+[TODO: Decide if and how the collector's request is authenticated.]
+
+The collector interacts with the leader to produce the final aggregate output.
+This process consists of a sequence of *collect requests* issued to the leader.
+Before a request can succeed, the aggregators must have verified and aggregated
+enough reports and the leader must have obtained the helper's encrypted output
+share (see {{pa-aggregate}}). In general, the procedure by which the aggregators
+verify and aggregate reports depends on parameters carried by the collect
+request. This procedure is described below in {{pa-aggregate}}.
+
+### Collect Request
+
+The collect protocol is an iterative procedure driven by the collector and
+parameterized by a PAParam. At a high level, it proceeds as follows. First, the
+collector initializes local per-protocol state using the corresponding
+PAParam. This state object has the following member functions:
+
+- CreateRequest(): Creates a PACollectReq object, defined below, that is sent to
+  the leader to complete one iteration of the protocol. Along with any
+  protocol-specific parameters, the message specifies a time interval
+  `[batch_start, batch_end)` that determines the batch of reports to be
+  aggregated. It must be that `batch_end - batch_start >= PAParam.batch_window`
+  and `batch_start` and `batch_end` are multiples of `PAParam.batch_window`.
+
+~~~
+struct {
+  PATask task;
+  Url helper;
+  uint64 batch_start; // The beginning of the batch in UNIX time.
+  Uint64 batch_end;   // The end of the batch in UNIX time (exclusive).
+  PAProto proto;
+  select (PACollectReq.proto) {
+    case prio: PrioCollectReq;
+    case hits: HitsCollectReq;
```

Comment:
Do we understand what the contents of this might be? Are they really going to be per-request? In particular, why do we need ```proto```. Each task should just have one proto

Reply:
- PrioCollectReq would minimally have some identifier for the validity circuit and, potentially, the field.
- HitsCollectReq might might specify one or two fields.

Reply:
I don't see why any of this needs to be per-request. They should just be fixed per-task.

Reply:
For Hits this is also going to convey the set of candidate prefixes. It's therefore not fixed per request.

Reply:
Well, we should definitely factor out the things that are per-task and things that are per-collector-request.